File: Python/ProblemSolving/Designer PDF Viewer.py
#!/bin/python3

import math
import os
import random
import re
import sys

def designerPdfViewer(h, word):
    maxi = -1
    for i in word:
        index = ord(i) - 97
        if h[index] > maxi:
            maxi = h[index]
    return maxi * len(word)

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    h = list(map(int, input().rstrip().split()))

    word = input()

    result = designerPdfViewer(h, word)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Scala/ProblemSolving/Designer PDF Viewer.scala
import java.io._
import scala.io._
import scala.math._

object Result {

  def designerPdfViewer(h: Array[Int], word: String): Int = {
    var maxi = -1
    for (i <- word){
      val wordAsciiNum = i.toInt
      val index = wordAsciiNum - 97
      if(h(index) > maxi){
        maxi = h(index)
      }
    }
    word.length * maxi
  }

}

object Solution {
    def main(args: Array[String]) {
        val printWriter = new PrintWriter(sys.env("OUTPUT_PATH"))

        val h = StdIn.readLine.replaceAll("\\s+$", "").split(" ").map(_.trim.toInt)

        val word = StdIn.readLine

        val result = Result.designerPdfViewer(h, word)

        printWriter.println(result)

        printWriter.close()
    }
}
